Sources/MapboxMaps/Style/Types/StyleColor.swift

```diff
@@ -0,0 +1,138 @@
+
+    /// A value from 0 to 1, as required by the Mapbox Style Spec
+    public let alpha: Double
+
+    /// Creates a `StyleColor` from individually-provided color components. Returns nil
+    /// if any of the channel values are out of the supported ranges.
+    /// - Parameters:
+    ///   - red: A value from 0 to 255, as required by the Mapbox Style Spec
+    ///   - green: A value from 0 to 255, as required by the Mapbox Style Spec
+    ///   - blue: A value from 0 to 255, as required by the Mapbox Style Spec
+    ///   - alpha: A value from 0 to 1, as required by the Mapbox Style Spec
+    public init?(red: Double, green: Double, blue: Double, alpha: Double) {
+        guard [red, green, blue].allSatisfy((0.0...255.0).contains),
+              (0.0...1.0).contains(alpha) else {
+            return nil
+        }
+        self.red = red
+        self.green = green
+        self.blue = blue
+        self.alpha = alpha
+    }
+
+    // MARK: - UIColor
+
+    /// Creates a `StyleColor` from a `UIColor`
+    /// - Parameter color: A `UIColor` in the sRGB color space
+    public init(_ color: UIColor) {
+        var red: CGFloat = 0.0
+        var green: CGFloat = 0.0
+        var blue: CGFloat = 0.0
+        var alpha: CGFloat = 0.0
+        guard color.getRed(&red, green: &green, blue: &blue, alpha: &alpha),
+              [red, green, blue, alpha].allSatisfy((0.0...1.0).contains) else {
+            fatalError("Please use a color in the sRGB color space")
+        }
+        self.red = Double(red * 255)
+        self.green = Double(green * 255)
+        self.blue = Double(blue * 255)
+        self.alpha = Double(alpha)
+    }
+
+    // MARK: - Expression
+
+    /// Creates a `StyleColor` from an `Expression`. Returns nil if
+    /// the expression operator is not `rgba`, if the expression does not
+    /// have four number-type arguments, or if the arguments are out of
+    /// the supported ranges.
+    /// - Parameter expression: An rgba expression
+    internal init?(expression: Expression) {
+        guard case .rgba = expression.operator,
+              expression.arguments.count == 4,
+              case .number(let red) = expression.arguments[0],
+              case .number(let green) = expression.arguments[1],
+              case .number(let blue) = expression.arguments[2],
+              case .number(let alpha) = expression.arguments[3] else {
+            return nil
+        }
+        self.init(red: red, green: green, blue: blue, alpha: alpha)
+    }
+
+    // MARK: - RGBA String
+
+    /// Creates a `StyleColor` from an rgba color string as defined
+    /// in the Mapbox Style Spec. Returns nil if the string is not a valid
+    /// rgba color string.
+    /// - Parameter rgbaString: An rgba color string
+    internal init?(rgbaString: String) {
+        let nsString = NSString(string: rgbaString)
+        let regex = try! NSRegularExpression(pattern: "rgba\\( *([0-9.]+) *, *([0-9.]+) *, *([0-9.]+) *, *([0-9.]+) *\\)", options: [])
```

Comment:
should clamp to the beginning and end of the input string, allowing leading and trailing whitespace